lib/Conversion/HWToLLVM/HWToLLVM.cpp

```diff
@@ -198,6 +198,38 @@ struct ArraySliceOpConversion
 };
 } // namespace
 
+//===----------------------------------------------------------------------===//
+// Explosion operation conversions
+//===----------------------------------------------------------------------===//
+
+namespace {
+/// Convert a StructExplodeOp to a LLVM dialect.
+/// Pattern: struct_explode(input) =>
+///          struct_extract(input, structElements_index(index)) ...
+struct StructExplodeOpConversion
+    : public ConvertOpToLLVMPattern<hw::StructExplodeOp> {
+  using ConvertOpToLLVMPattern<hw::StructExplodeOp>::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(hw::StructExplodeOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Type inputTy = typeConverter->convertType(op.getInput().getType());
+
+    for (size_t i = 0, e = inputTy.cast<LLVM::LLVMStructType>().getBody().size();
+         i < e; ++i) {
+
+      Value castInput = typeConverter->materializeTargetConversion(
+          rewriter, op.getInput().getLoc(), inputTy, op.getInput());
+
+      rewriter.create<LLVM::ExtractValueOp>(op->getLoc(), castInput, i);
+    }
+    rewriter.eraseOp(op);
+    return success();
```

Comment:
When try to firgure out the way to do conversion, found some operations can not be called by the **op** object which was derived from StructExplodeOp class, so I have to change the writing way to get over the problem, Otherwise, compilation will fail.So what definitions determine the methods of operation that they can use, is it in TD dialects file or some others...?